mlir/include/mlir/Dialect/Bufferization/IR/BufferizableOpInterface.h

```diff
@@ -578,6 +578,81 @@ class AnalysisState {
       insideMutuallyExclusiveRegionsCache;
 };
 
+/// BufferizationState provides information about the state of the IR during the
+/// bufferization process.
+class BufferizationState {
+public:
+  /// Base class for BufferizationState extensions that allow BufferizationState
+  /// to contain user-specified information in the state object. The extension
+  /// mechanism of BufferizationState mirrors the one of OneShotAnalysisState.
+  class Extension {
```

Comment:
I think this entire extension mechanism is not needed anymore. Let's remove it for now, and bring it back if we find a use case for it in the future.